### ISSUE TITLE
When an object's non-null property resolver returns null (or exception), the 'null' isn't bubbled up to the parent

### ISSUE THREAD

**Issue by diaconesq** [graphql-dotnet/graphql-dotnet#424]
The spec says a **non-null** property that is resolved as null (or throws an error) should propagate up and make the whole object become null:
http://facebook.github.io/graphql/#sec-Errors

In the 'StarWars' example, if I change the `Human.Name` property to non-null in the schema, and query for 
```
{
  human(id: "IHaveNoName") {
    name
  }
}
```
and the corresponding object exists in the "database" but has the `Name` property set to `null`, the response should be:  
```
{
  "data": {
    "human": null
  },
  "errors": [
    {
      "message": "Error trying to resolve name." [...]
    }
  ]
}
```

but instead it is:

```
{
  "data": {
    "human": { // the null value should propagate to here
      "name": null //this is invalid, should never be null
    }
  },
  "errors": [
    {
      "message": "Error trying to resolve name." [...]
    }
  ]
}
```

**Comment by KoditkarVedant** [comment]
@joemcbride I was going through `graphql` spec for `List and NonNull` and noticed that as of now we do not handle it in out library.
```
myField: [String!]
```
This means that the list itself can be null, but it can't have any null members. For example, in JSON:
```
myField: null // valid
myField: [] // valid
myField: ['a', 'b'] // valid
myField: ['a', null, 'b'] // error
```
Now, let's say we defined a Non-Null List of Strings:
```
myField: [String]!
```
This means that the list itself cannot be null, but it can contain null values:
```
myField: null // error
myField: [] // valid
myField: ['a', 'b'] // valid
myField: ['a', null, 'b'] // valid
```
[ref](https://graphql.github.io/learn/schema/#lists-and-non-null)

@joemcbride should we track this as separate or should we consider as part of this enhancement itself. 
I am curious to know your thoughts.

**Comment by joemcbride** [comment]
@KoditkarVedant I would be pretty surprised if that is the case.  Lets track that separately and see if you can add some tests to show that its broken.

**Comment by KoditkarVedant** [comment]
@joemcbride Is it okay to keep those test cases failing or should I skip them from running?

**Comment by joemcbride** [comment]
@KoditkarVedant I think its fine to have them fail, then when the build passes we know we've fixed it.

**Comment by KoditkarVedant** [comment]
> 
> 
> @KoditkarVedant I think its fine to have them fail, then when the build passes we know we've fixed it.

Done.

**Comment by joemcbride** [comment]
This should be fixed with #980
